### PR TITLE
chore(python): remove usage of pydantic aliases

### PR DESCRIPTION
**Pull request by armandobelardo**
Pydantic field aliases are removed and replaced with an internal representation. This allows for more robust handling of field aliases and prevents issues with Pydantic V2 and mypy.

Previously, you'd have for V1 and V2 compatibility in Pydantic, you'd want to conditionally apply the config class within the base model, however this would lead to mypy errors when filling out a model with it's field alias. To solve this, We used the deprecated `class Config`, regardless of the Pydantic version to satisfy mypy, which lead to warnings in the console.

Now, we've removed the field aliases and replaced them with an internal representation, which allows us to avoid pydantic config altogether.

### CODE REVIEW

File: seed/python-sdk/imdb/src/seed/imdb/types/create_movie_request.py [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 from ...core.pydantic_utilities import IS_PYDANTIC_V2
 import typing
 import pydantic
 
 
 class CreateMovieRequest(UniversalBaseModel):
-    title: str
-    rating: float
+    title: typing_extensions.Annotated[str, FieldMetadata(alias="title")]
+    rating: typing_extensions.Annotated[float, FieldMetadata(alias="rating")]
```
Comment (dsinghvi) [inline seed/python-sdk/imdb/src/seed/imdb/types/create_movie_request.py:13]: @armandobelardo how hard is it to ignore adding an alias if the alias is the same as the normal value?

Comment (armandobelardo) [reply seed/python-sdk/imdb/src/seed/imdb/types/create_movie_request.py:13]: interesting, this shouldn't be happening, will take a look

Comment (armandobelardo) [reply seed/python-sdk/imdb/src/seed/imdb/types/create_movie_request.py:13]: image
 🤦

Comment (armandobelardo) [reply seed/python-sdk/imdb/src/seed/imdb/types/create_movie_request.py:13]: it's supposed to just be `if is_aliased`

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**